src/librustc/mir/mod.rs

```diff
@@ -1674,6 +1674,10 @@ impl<'tcx> Statement<'tcx> {
     /// Changes a statement to a nop. This is both faster than deleting instructions and avoids
     /// invalidating statement indices in `Location`s.
     pub fn make_nop(&mut self) {
+        // `Statement` contributes significantly to peak memory usage. Make
+        // sure it doesn't get bigger.
+        debug_assert!(mem::size_of::<Statement<'_>>() <= 56usize);
```

Comment:
Could you use a static assert instead:

```rust
#![feature(const_transmute, const_let)]

use std::mem;

#[allow(dead_code)]
const ASSERT_STATMENT_IS_56_BYTES: () = {
    unsafe { mem::transmute::<_, [u8; 56]>(Statement::Nop); }
    ()
};
```

Reply:
librustc also has a [`static_assert!`](https://github.com/rust-lang/rust/blob/945372d26818f93d6f5cded7b751749e280b67bf/src/librustc/macros.rs#L66) macro for this kind of pattern.

Reply:
Didn't know about that.

```rust
static_assert!(STATMENT_IS_AT_MOST_56_BYTES: std::mem::size_of::<Statement<'_>>() <= 56usize);
```